packages/core/src/services/health/service.ts

```diff
@@ -418,6 +419,156 @@ export class Service {
     return await this.getEVMNodeHealth(node, { harmony: true });
   };
 
+  /* ----- Near ----- */
+  private async getNEARBlockHeight(
+    url: string,
+  ) {
+    try {
+      const { data } = await this.rpc.post(
+        url,
+        { jsonrpc: '2.0', id: "dontcare", method: "status", params: [] },
+      );
+      return data.result.sync_info.latest_block_height;
+    } catch (error) {
+      const stringError = JSON.stringify(error);
+      throw new Error(
+        `getBlockHeight could not contact blockchain node ${stringError} ${url}`,
+      );
+    }
+  };
+
+  private getNEARNodeHealth = async (
+    node: INode,
+  ): Promise<IHealthResponse> => {
+    const { name, chain, url, host, port } = node;
+    const { allowance } = chain;
+
+    let healthResponse: IHealthResponse = {
+      name,
+      status: EErrorStatus.OK,
+      conditions: EErrorConditions.HEALTHY,
+    };
+
+    /* Check if node is online and RPC up */
+    const isNodeListening = await this.isNodeListening({ host: host.ip, port });
+    if (!isNodeListening) {
+      return {
+        ...healthResponse,
+        status: EErrorStatus.ERROR,
+        conditions: EErrorConditions.OFFLINE,
+      };
+    }
+    const NEARBlockHeight = await this.getNEARBlockHeight(url);
+    if (!NEARBlockHeight) {
+      return {
+        ...healthResponse,
+        status: EErrorStatus.ERROR,
+        conditions: EErrorConditions.NO_RESPONSE,
+      };
+    }
+    let referenceUrls: IReferenceURL[];
+    const { healthyOracles, badOracles } = await this.getOracles(chain);
+    if (healthyOracles.length >= 1) {
+      referenceUrls = healthyOracles;
+    } else {
+      const healthyPeers = await this.getPeers(chain, node.id);
+      if (!healthyOracles.length && healthyPeers.length < 2) {
+        return {
+          ...healthResponse,
+          status: EErrorStatus.ERROR,
+          conditions: EErrorConditions.NO_PEERS,
+        };
+      } else if (!healthyOracles.length && healthyPeers.length >= 2) {
+        healthResponse = {
+          ...healthResponse,
+          details: { noOracle: true, numPeers: healthyPeers.length },
+        };
+      }
+      referenceUrls = [...healthyOracles, ...healthyPeers];
+    }
+
+    if (badOracles.length) {
+      healthResponse = {
+        ...healthResponse,
+        details: {
+          ...healthResponse.details,
+          badOracles: badOracles.map(({ url }) => url),
+        },
+      };
+    }
+
+
+    try {
+      /* Get node's block height, highest block height from reference nodes */
+
+      let externalHeights = await Promise.all(
+        referenceUrls.map(async ref => {
```

Comment:
Don't need async here since .map doesn't work async so will just return an array of Promises.

Also, just curious since `ref` inside the .map function doesn't have ( ) around it, do you have Prettier configured? It should add those brackets by default on autosave. I know it seems like a minor thing but if our code style is inconsistent it leads to annoying PRs cus they are full of little changes caused by Prettier.

Reply:
the async part was a typo, when I was trying to update to `Promise.all`. I will update that.

The `ref` in the `.map` does not have () because I am using the explicit `return` statement, if it were the implicit es6 return, I won't use a `return` statement, then will use the brackets (). I am developing on the node-nanny remote server I think the prettier is properly configured, I want to assume it is configured on the server; But then it is not a prettier issue it's because I used the explicit return in the arrow function. 

On the style of code, I think there will always be these little differences in style when multiple people are collaborating, especially if there are no strict guides.  But will update the PR

Reply:
Yeah, makes sense. Looks good man,  thanks for the update I'm going to merge it now.